src/overlays/actors/ovl_En_Ending_Hero6/z_en_ending_hero6.c

```diff
@@ -24,20 +55,133 @@ const ActorInit En_Ending_Hero6_InitVars = {
     (ActorFunc)EnEndingHero6_Draw,
 };
 
-#endif
+static FlexSkeletonHeader* D_80C24200[] = { &D_0600B0CC, &D_06007908, &D_06007150, &D_0600D640, &D_0600A850,
+                                            &D_0600A850, &D_0600A850, &D_0600A850, &D_0600A850 };
+
+static AnimationHeader* D_80C24224[] = { &D_06000BE0, &D_060011C0, &D_06000E50, &D_06002A84, &D_06002FA0,
+                                         &D_06002FA0, &D_06002FA0, &D_06002FA0, &D_06002FA0 };
+
+static s32 D_80C24248[] = { 15, 20, 17, 17, 17, 17, 17, 17, 17 };
+
+void EnEndingHero6_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero6* this = THIS;
+
+    this->actor.colChkInfo.mass = 255;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, D_80C24200[this->npcIndex], D_80C24224[this->npcIndex],
+                     this->limbDrawTable, this->transitionDrawTable, D_80C24248[this->npcIndex]);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    func_80C23DDC(this);
+}
+
+void EnEndingHero6_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80C23D60(EnEndingHero6* this, s32 npcIndex) {
+    this->animIndex = npcIndex;
+    this->frameCount = SkelAnime_GetFrameCount(&D_80C24224[npcIndex]->common);
+    SkelAnime_ChangeAnim(&this->skelAnime, D_80C24224[this->animIndex], 1.0f, 0.f, this->frameCount, 0, 0.0f);
+}
+
+void func_80C23DDC(EnEndingHero6* this) {
+    func_80C23D60(this, this->npcIndex);
+    this->unk290 = 1;
+    this->actionFunc = func_80C23E18;
+}
+
+void func_80C23E18(EnEndingHero6* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnEndingHero6_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero6* this = THIS;
+
+    if (this->unk286 != 0) {
+        this->unk286--;
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    if (this->unk288 == 0) {
+        this->eyeState++;
+        if (this->eyeState >= 3) {
+            this->eyeState = 0;
+            this->unk288 = (s16)Rand_ZeroFloat(60.0f) + 20;
+        }
+    }
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1DU);
+}
+
+void func_80C23F14(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
```

Comment:
This function is a very common pattern, and has a particular name: `PostLimbDraw`, so this one is `EnEndingHero6_PostLimbDraw`. (It is used to draw extra things on the model after the limb positions are updated, hence the name.)